src/xmagics/xassist.cpp

```diff
@@ -60,6 +60,80 @@ namespace xcpp
         }
     };
 
+    class model_manager
+    {
+    public:
+
+        static void save_model(const std::string& model, const std::string& model_name)
+        {
+            std::string model_file_path = model + "_model.txt";
+            std::ofstream out(model_file_path);
+            if (out)
+            {
+                out << model_name;
+                out.close();
+                std::cout << "Model saved for model " << model << std::endl;
+            }
+            else
+            {
+                std::cerr << "Failed to open file for writing model for model " << model << std::endl;
+            }
+        }
+
+        static std::string load_model(const std::string& model)
+        {
+            std::string model_file_path = model + "_model.txt";
+            std::ifstream in(model_file_path);
+            std::string model_name;
+            if (in)
+            {
+                std::getline(in, model_name);
+                in.close();
+                return model_name;
+            }
+
+            std::cerr << "Failed to open file for reading model for model " << model << std::endl;
+            return "";
+        }
+    };
+
+    class url_manager
+    {
+    public:
+
+        static void save_url(const std::string& model, const std::string& url)
+        {
+            std::string url_file_path = model + "_url.txt";
+            std::ofstream out(url_file_path);
+            if (out)
+            {
+                out << url;
+                out.close();
+                std::cout << "URL saved for model " << model << std::endl;
+            }
+            else
+            {
+                std::cerr << "Failed to open file for writing URL for model " << model << std::endl;
+            }
+        }
+
+        static std::string load_url(const std::string& model)
+        {
+            std::string url_file_path = model + "_url.txt";
+            std::ifstream in(url_file_path);
+            std::string url;
+            if (in)
+            {
+                std::getline(in, url);
+                in.close();
+                return url;
+            }
+
+            std::cerr << "Failed to open file for reading URL for model " << model << std::endl;
+            return "";
+        }
+    };
```

Comment:
Technically any utility function like these should be tested for (either in test_interpreter.cpp itself or maybe through a test_utils.cpp if possible) 

But that can be done through subsequent PRs (an issue can be opened to track the same) !